tests/unit/dojson/test_dojson_hepnames.py

```diff
@@ -623,13 +628,39 @@ def test_other_names_from_400__triple_a():
 
 
 def test_advisors(marcxml_to_json, json_to_marc):
-    assert (marcxml_to_json['advisors'][0]['name'] ==
-            json_to_marc['701'][0]['a'])
-    assert (marcxml_to_json['advisors'][0]['_degree_type'] ==
-            json_to_marc['701'][0]['g'])
+    snippet = {
+        'advisors':[
+            {
+                'name': 'Rivelles, Victor O.',
+                'degree_type': 'phd',
+                '_degree_type': 'PhD',
+                'record': {
+                    '$ref': 'http://localhost:5000/api/authors/991627',
+                },
+                'curated_relation': True
+            }
+        ]
+    }
+    expected = {
+        '701': [
+            {
+                'a': 'Rivelles, Victor O.',
+                'g': 'phd',
+                'x': 991627,
+                'y': '1'
+            }
+        ]
+    }
+
+    result = hepnames2marc.do(snippet)
+
+    assert expected == result
```

Comment:
I changed the test in this way for 2 reasons:

1. It is more readable
2. In order to test all the 701 fields